frontend/src/features/admin-form/create/builder-and-design/mutations/useDeleteFormField.ts

```diff
@@ -137,6 +140,38 @@ export const useDeleteFormField = () => {
     },
   )
 
+  const handleDeleteMultipleFieldsSuccess = useCallback(
+    (_data: unknown, fieldIds: string[]) => {
+      queryClient.setQueryData<AdminFormDto>(adminFormKey, (oldForm) => {
+        // Should not happen, should not be able to update field if there is no
+        // existing data.
+        if (!oldForm) throw new Error('Query should have been set')
+        const deletedFieldIndices = fieldIds
+          .map((fieldId) =>
+            oldForm.form_fields.findIndex((ff) => ff._id === fieldId),
+          )
+          .filter((index) => index >= 0)
+        oldForm.form_fields = oldForm.form_fields.filter(
+          (_field, index) => !deletedFieldIndices.includes(index),
```

Comment:
Interesting optimisation here to reduce the scope of `deletedFieldIndicies` was confused at first. 😅